waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandlerFactory.java

```diff
@@ -63,15 +64,12 @@ public CloseableIHMSHandler create() {
   }
 
   private MappingEventListener createDatabaseMappingService() {
-    switch (waggleDanceConfiguration.getDatabaseResolution()) {
-    case MANUAL:
+    if (waggleDanceConfiguration.getDatabaseResolution() == DatabaseResolution.MANUAL) {
```

Comment:
The switch reads better, thanks.

Reply:
If I put back the switch, then we can't test the default branch. So if `getDatabaseResolution` returns `null`, the switch throws an NPE.

Reply:
Ah, ok, was worth a shot!